src/coreclr/jit/assertionprop.cpp

```diff
@@ -164,6 +164,22 @@ bool IntegralRange::Contains(int64_t value) const
         case GT_CAST:
             return ForCastOutput(node->AsCast());
 
+#ifdef FEATURE_HW_INTRINSICS
+        case GT_HWINTRINSIC:
+            if (HWIntrinsicInfo::IsXCNTIntrinsic(node->AsHWIntrinsic()->GetHWIntrinsicId()))
+            {
+                return {SymbolicIntegerValue::Zero, SymbolicIntegerValue::IntMax};
```

Comment:
@EgorBo, is it worth modeling these more accurately?

Most of these are actually `0-7`, `0-15`, `0-31`, or `0-63` depending on the type.

Reply:
Yes, I was wondering the same thing! Would UByte would probably be a more appropriate upper bounds for the foreseeable future 😆

Reply:
It is, but `IntegralRange` cannot do that today (but will be able to in the future). These should use `SymbolicIntegerValue::ByteMax` for now and have a `TODO-Casts: specify more precise ranges once "IntegralRange" supports them`.

Reply:
Updated in 535c0474d0e7079d4d90b9eaa3f499e0743c73fc